examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+#pragma once
+
+#include "EQDiagram.hpp"
+#include "ParametricEqControls.hpp"
+
+#include <DspFilters/Butterworth.h>
+#include <DspFilters/ChebyshevI.h>
+#include <DspFilters/Filter.h>
+#include <halp/custom_widgets.hpp>
+#include <halp/layout.hpp>
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <string_view>
+
+namespace examples
+{
+// ********************************************************************************
+//                                  Actual EQ
+// ********************************************************************************
+struct ParametricEq
+{
+  static constexpr auto name() { return "Parametric Equalizer"; }
+  static constexpr auto c_name() { return "ngry_parameq"; }
+  static constexpr auto author() { return "Nicolas Gry"; }
+  static constexpr auto category() { return "Demo"; }
+  static constexpr auto description()
+  {
+    return "Parametric Equalizer, using Butterworth equations for all filters except "
+           "the bandshelf filter (which is computed with Chebyshev Type I equations).";
+  }
+  static constexpr auto uuid() { return "ec972d12-edba-4a9c-9736-9176efad17e3"; }
+
+  static constexpr int bandsNb = 8;  // EQ Bands number
+  static constexpr int maxOrder = 7; //Filters maximum order
+  struct ins
+  {
+    struct
+    {
+      static constexpr auto name() { return "Input"; }
+      const double** samples;
+      int channels;
+      float sampleRate;
+    } audio;
+
+    EQBand band1, band2, band3, band4, band5, band6, band7, band8;
```

Comment:
this is giving me motivation to make this simpler aha, sorry you had to go through this

Reply:
i eventually worked things around but it'd be definitely nice if the ui of the bands could change on the run depending on the filter type aha